corehq/apps/reports/models.py

```diff
@@ -657,25 +664,47 @@ def hour_choices(cls):
         """Tuples for hour number and human-readable hour"""
         return tuple([(val, "%s:00" % val) for val in range(24)])
 
-    def send(self):
-        from dimagi.utils.django.email import send_HTML_email
-        from corehq.apps.reports.views import get_scheduled_report_response
+    @property
+    @memoized
+    def recipients_by_language(self):
+        user_languages = {
+            user['username']: user['language']
+            for user in get_user_docs_by_username(self.all_recipient_emails)
+            if 'username' in user and 'language' in user
+        }
+        fallback_language = user_languages.get(self.owner_email, 'en')
+
+        recipients = defaultdict(list)
+        for email in self.all_recipient_emails:
+            language = user_languages.get(email, fallback_language)
+            recipients[language].append(email)
+        return recipients
 
+    def send(self):
         # Scenario: user has been removed from the domain that they
         # have scheduled reports for.  Delete this scheduled report
         if not self.owner.is_member_of(self.domain):
             self.delete()
             return
 
-        if self.all_recipient_emails:
-            title = "Scheduled report from CommCare HQ"
-            if hasattr(self, "attach_excel"):
-                attach_excel = self.attach_excel
-            else:
-                attach_excel = False
-            body, excel_files = get_scheduled_report_response(self.owner, self.domain, self._id, attach_excel=attach_excel)
-            for email in self.all_recipient_emails:
-                send_HTML_email(title, email, body.content, email_from=settings.DEFAULT_FROM_EMAIL, file_attachments=excel_files)
+        if self.recipients_by_language:
+            for language, emails in self.recipients_by_language.items():
+                self._get_and_send_report(language, emails)
```

Comment:
I think it's unavoidable to generate the report more than once - this does it once per language, which I imagine will be ~1 per report anyways, so it's probably fine.